docs: rewrite Java template tutorial to AsyncAPI v3 (#2029)
asyncapi/generator: apps/generator/docs/generator-template-java.md

    "version": "0.0.1",
    "description": "A template that generates a Java MQTT client using MQTT.",
    "generator": {

>       "apiVersion": "v2",
>       "generator": ">=2.0.0 <3.0.0",

Reviewer: AsyncAPI 3.0.0 documents require Parser-API v3. With `v2` the parser will reject the doc. Python tutorial uses `v3` and `>=2.0.0 <4.0.0`.

```suggestion
      "apiVersion": "v3",
      "generator": ">=2.0.0 <4.0.0",
```

    "version": "0.0.1",
    "description": "A template that generates a Java MQTT client using MQTT.",
    "generator": {

>       "apiVersion": "v2",
>       "generator": ">=2.0.0 <3.0.0",

Reviewer: Same fix as the earlier `package.json` block — keep both copies consistent so a reader copying either one ends up with a working template.

```suggestion
      "apiVersion": "v3",
      "generator": ">=2.0.0 <4.0.0",
```

}
  
  /*
   * This function returns a list of objects, one for each receive operation.
   * name - holds information about the `operationId` defined in the AsyncAPI document
   * topic - holds information about the topic's address.
   *
   * It requires operations from the parsed AsyncAPI v3 document.
   */
  function getTopicsFromOperations(operations) {

>     const receiveOperations = operations.filterBySend();

Reviewer: `operations` if you look into line 381 are already filtered correct by receive, so not filter by send again makes no sense, will always be empty, no?

>     const receiveOperations = operations.filterBySend();

Reviewer: All operations in the example are `action: receive`, so `filterBySend()` returns empty and no `send*` methods are generated. Variable name and surrounding comment also say "receive".

```suggestion
    const receiveOperations = operations.filterByReceive();
```

    let topicsDetails = []
  
    receiveOperations.forEach((op) => {
      const topic = {}
      const operationId = op.id()
      topic.name = operationId.charAt(0).toUpperCase() + operationId.slice(1)

>       topic.topic = op.channel().address()

Reviewer: v3 Operations expose `channels()` (a collection), not `channel()` (only `OperationReply` has singular). Throws at runtime.

```suggestion
      topic.topic = op.channels()[0].address()
```

Import the `TopicFunction` component in your template code in **index.js** and add the template code to generate the functions for the topics which the `Temperature Service` application is subscribed to. In your case, the final version of your template code should look like this:

```js
import { File } from '@asyncapi/generator-react-sdk';
import { TopicFunction } from '../components/TopicFunction'

export default function ({ asyncapi, params }) {
    // v3: Get receive operations instead of channels
    const operations = asyncapi.operations();  

>     const server = asyncapi.server(params.server);

Reviewer: `asyncapi.server()` (singular) does not exist on the v3 document model — only `servers()`. This will throw `is not a function`.

```suggestion
    const server = asyncapi.servers().get(params.server);
```

>     // v3: Get receive operations instead of channels
>     const operations = asyncapi.operations();  
>     const server = asyncapi.server(params.server);
>     // Generate Java code for each topic dynamically using TopicFunction
>     const topicMethods = TopicFunction({ operations });  // This will return Java methods as text

Reviewer: The comment claims we "get receive operations" but the code passes *all* operations unfiltered, then `TopicFunction` re-filters internally. Either filter here (mirrors the Python tutorial) or drop the misleading comment.

```suggestion
    const operations = asyncapi.operations().filterByReceive();
    const server = asyncapi.servers().get(params.server);
    // Generate Java code for each topic dynamically using TopicFunction
    const topicMethods = TopicFunction({ operations });  // This will return Java methods as text
```

servers:
  dev:
    host: test.mosquitto.org:1883
    protocol: mqtt

channels:
  temperature/dropped:    
    address: temperature/dropped
  temperature/risen:
    address: temperature/risen

operations:
  temperatureDrop:
    action: receive
    channel:
      $ref: '#/channels/temperature~1dropped'
  temperatureRise:
    action: receive
    channel:
      $ref: '#/channels/temperature~1risen'

> 

Reviewer: This doc is significantly stripped down vs. the Python tutorial: no `messages`, no channel `description`, no operation `summary`, no `components.schemas`. Students learning from the Java page see a much thinner spec than the Python page. Recommend mirroring the Python tutorial's §5d doc and using clean channel keys (`temperatureDropped`, `temperatureRisen`) instead of slash-keys that force `~1` escaping in `$ref`.
